lib/location/method_gnss.c

```diff
@@ -866,6 +871,113 @@ static void method_gnss_print_pvt(const struct nrf_modem_gnss_pvt_data_frame *pv
 	}
 }
 
+#if defined(CONFIG_LOCATION_SERVICE_NRF_CLOUD_GNSS_POS_CLOUD_SEND)
+
+static int method_gnss_nrf_cloud_data_send(char *body)
+{
+	int ret;
+
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+	struct nrf_cloud_tx_data mqtt_msg = {
+		.data.ptr = body,
+		.data.len = strlen(body),
+		.qos = MQTT_QOS_1_AT_LEAST_ONCE,
+		.topic_type = NRF_CLOUD_TOPIC_MESSAGE,
+	};
+
+	ret = nrf_cloud_send(&mqtt_msg);
+	if (ret) {
+		LOG_ERR("MQTT: location data sending failed: %d", ret);
+	}
+#elif defined(CONFIG_NRF_CLOUD_COAP)
+	ret = nrf_cloud_coap_json_message_send(body, false, true);
+	if (ret) {
+		LOG_ERR("CoAP: location data sending failed");
+	}
+#elif defined(CONFIG_NRF_CLOUD_REST)
+#define REST_DETAILS_RX_BUF_SZ 300 /* No payload in a response, "just" headers */
+	static char rx_buf[REST_DETAILS_RX_BUF_SZ];
+	static char device_id[NRF_CLOUD_CLIENT_ID_MAX_LEN + 1];
+	static struct nrf_cloud_rest_context rest_ctx = {
+		.connect_socket = -1,
+		.keep_alive = false,
+		.rx_buf = rx_buf,
+		.rx_buf_len = sizeof(rx_buf),
+		.fragment_size = 0
+	};
+
+	ret = nrf_cloud_client_id_get(device_id, sizeof(device_id));
+	if (ret == 0) {
+		ret = nrf_cloud_rest_send_device_message(&rest_ctx, device_id, body, false, NULL);
+		if (ret) {
+			LOG_ERR("REST: location data sending failed: %d", ret);
+		}
+	} else {
+		LOG_ERR("Failed to get device ID, error: %d", ret);
+	}
+#endif
+	return ret;
+}
+
+static void method_gnss_nrf_cloud_pos_send(struct location_data *location_result)
+{
+	int err = 0;
+	char *json_str = NULL;
+	cJSON *gnss_data_obj = NULL;
+	struct nrf_cloud_gnss_data gnss_data = {
+		.type = NRF_CLOUD_GNSS_TYPE_PVT,
+		.pvt = {
+			.lon	  = location_result->longitude,
+			.lat	  = location_result->latitude,
+			.accuracy = location_result->accuracy,
+		}
+	};
+	const struct tm time = {
+		.tm_year = location_result->datetime.year - 1900,
+		.tm_mon = location_result->datetime.month - 1,
+		.tm_mday = location_result->datetime.day,
+		.tm_hour = location_result->datetime.hour,
+		.tm_min = location_result->datetime.minute,
+		.tm_sec = location_result->datetime.second,
+	};
+
+	gnss_data.ts_ms = timeutil_timegm64(&time) * 1000 + location_result->datetime.ms;
+
+	gnss_data_obj = cJSON_CreateObject();
+	if (gnss_data_obj == NULL) {
+		LOG_ERR("Failed to encode GNSS position data, out of memory");
+		err = -ENOMEM;
+		goto cleanup;
+	}
+
+	/* Encode the GNSS location data */
+	err = nrf_cloud_gnss_msg_json_encode(&gnss_data, gnss_data_obj);
```

Comment:
would be nice to use `nrf_cloud_coap_location_send` if we're using the CoAP backend otherwise.